File: .py/Program/PC.py
from Computer import Computer
from PSU import PSU

class PC (Computer):
    __PSU = None

    def __init__ (self, brand = "", model = "", OS = "", CPU = None, RAM = None, Storage = None, PSU = None):
        super ().__init__ (brand, model, OS, CPU, RAM, Storage)
        self.__PSU = PSU

    def setPSU (self, PSU):
        self.__PSU = PSU

    def getPSU (self):
        return self.__PSU
    
    def info (self):
        print ("| PC")
        super ().info ()
        psu = self.getPSU ()
        if psu:
            print (f"| PSU               : { psu.getBrand() } { psu.getModel() } { psu.getWattage() }W { psu.getFormFactor() } { psu.getEfficiency() } { psu.getModular() } Modular")

File: .cpp/Program/PC.cpp
#pragma once
#include <bits/stdc++.h>
#include "Computer.cpp"
#include "PSU.cpp"
using namespace std;

class PC : public Computer {
    private:
        PSU psu;

    public:
        PC () { }

        PC (string brand, string model, string OS, CPU cpu, vector<RAM> ram, vector<Storage*> storage, PSU psu) : Computer (brand, model, OS, cpu, ram, storage) {
            this->psu = psu;
        }

        void setPSU (PSU psu) { this->psu = psu; }
        
        PSU getPSU () { return psu; }

        void info () override {
            cout << "PC" << endl;
            Computer::info();
            cout << "| PSU               : " << psu.getBrand() << " " << psu.getModel() << " " << psu.getWattage() << "W " << psu.getFormFactor() << " " << psu.getEfficiency() << " " << psu.getModular() << " Modular" << endl;
        }

        ~PC () { }
};